defaultclient.go

```diff
@@ -488,9 +488,11 @@ func (client *DefaultClient) ValidatePermission(claims *JWTClaims,
 			return false, err
 		}
 
-		grantedRolePermissions = client.applyUserPermissionResourceValues(grantedRolePermissions, claims,
+		replacedGrantedRolePermissions := make([]Permission, len(grantedRolePermissions))
```

Comment:
maybe better to move these codes inside `client.applyUserPermissionResourceValues`:

`
replacedGrantedRolePermissions := make([]Permission, len(grantedRolePermissions))
copy(replacedGrantedRolePermissions, grantedRolePermissions)
`

As the `client.applyUserPermissionResourceValues` also being called in the line `542` below.

Reply:
I think the problem happens here: https://github.com/AccelByte/iam-go-sdk/commit/0f0a649ff6071ec88dd58f74dd22dc6e60b0347c#r146542653